lib/client.js

```diff
@@ -50,28 +49,98 @@ class CloudantClient {
     // build plugin array
     var plugins = [];
     if (typeof this._cfg.plugin === 'undefined' && typeof this._cfg.plugins === 'undefined') {
-      plugins = ['cookieauth']; // default
+      plugins = [ 'cookieauth' ]; // default
     } else {
       [].concat(self._cfg.plugin).concat(self._cfg.plugins).forEach(function(plugin) {
-        if (typeof plugin === 'function' && !(plugin._pluginVersion >= 2)) {
-          if (self.useLegacyPlugin) {
-            throw new Error('Using multiple legacy plugins in not supported');
-          } else {
-            debug('Using legacy plugin.');
-            self.useLegacyPlugin = true;
-            client = plugin; // use legacy plugin as client
-            return;
-          }
+        if (typeof plugin !== 'function' || plugin.pluginVersion >= 2) {
+          plugins.push(plugin);
+        } else if (self.useLegacyPlugin) {
+          throw new Error('Using multiple legacy plugins in not supported');
+        } else {
+          self.useLegacyPlugin = true;
+          client = plugin; // use legacy plugin as client
         }
-        plugins.push(plugin);
       });
     }
 
     // initialize the internal client
     self._initClient(client);
 
     // add plugins
-    self.addPlugins(plugins);
+    self._addPlugins(plugins);
+  }
+
+  _addPlugins(plugins) {
```

Comment:
This is now private. Although new plugins could be added after construction, the legacy plugins can't as they overwrite the underlying request client being used.